### PR TITLE
Automl refactor

### PR DESCRIPTION
**Pull request by nnegrey**
Ditch this PR: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/2470
So that the samples go in the right place from the get-go.

Once merged, I will update docs and delete the branches

### CODE REVIEW

File: automl/cloud-client/language_sentiment_analysis_model_management_text.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python
```
Comment (davidcavazos) [inline automl/cloud-client/language_sentiment_analysis_model_management_text.py]: language_sentiment_analysis_model_management_***test***.py ?

Comment (nnegrey) [reply automl/cloud-client/language_sentiment_analysis_model_management_text.py]: Oops

File: automl/cloud-client/get_model_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    print(u'Model evaluation name: {}'.format(response.name))
+    print(u'Model annotation spec id: {}'.format(response.annotation_spec_id))
+    print('Create Time:')
+    print(u'\tseconds: {}'.format(response.create_time.seconds))
+    print(u'\tnanos: {}'.format(response.create_time.nanos / 1e9))
+    print(u'Evaluation example count: {}'.format(
+        response.evaluated_example_count))
+    print(u'Model evaluation metrics: {}'.format(
+        response.translation_evaluation_metrics))
```
Comment (sirtorry) [inline automl/cloud-client/get_model_evaluation.py]: this should be different for each flavour. please add a couple start/end region tags.

Comment (sirtorry) [reply automl/cloud-client/get_model_evaluation.py]: alternatively, you can use response.metrics

File: automl/cloud-client/delete_model.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    # [START automl_delete_model]
+    from google.cloud import automl
+
+    # TODO(developer): Uncomment and saet the following variables
+    # project_id = 'YOUR_PROJECT_ID'
+    # model_id = 'YOUR_MODEL_ID'
+
+    client = automl.AutoMlClient()
+    # Get the full path of the model.
+    model_full_id = client.model_path(project_id, 'us-central1', model_id)
```
[the remark this answers is not in the document]
Comment (nnegrey) [reply automl/cloud-client/delete_model.py]: To do that, it requires more than just changing the location, checkout the samples here, where we have specific samples demonstrating how to do this: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/2497/files

File: automl/cloud-client/get_model_evaluation.py (hunk 2 above)
Comment (nnegrey) [reply automl/cloud-client/get_model_evaluation.py]: Oops, done. 
metrics won't work, it is a `oneof` field.

**Comment by davidcavazos** [comment]
I don't know how you would feel about this, but in the [Go samples](https://github.com/GoogleCloudPlatform/golang-samples/tree/master/automl) we created separate directories for the language, translate and vision samples and that helped organize them a little better. It removed a lot of the name clutter and was easier to navigate.

We also named them as `dataset_*.go`, `model_*.go`, etc. to make it easier for people to find all the methods related to each resource. Within them, we still kept the actual function names with the verb first for readability (like `createDataset`, createModel`). It is a little inconsistent in terms of file name and function name, but I think it makes it easier to read and navigate from GitHub overall.

**Comment by nnegrey** [comment]
At this point, I don't want to move anything that might break docs, but later that would probably be a good idea.

**Comment by nnegrey** [comment]
@leahecole, Kokoro seems to be having some build issues or something. Just FYI

**Comment by kurtisvg** [comment]
@nnegrey Please ignore the Kokoro tests for now - we're still working on them :)

As long as the two required tests pass you should be okay.

File: automl/cloud-client/get_operation_status.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def get_operation_status(operation_full_id):
+    """Get operation status."""
+    # [START automl_get_operation_status]
+    from google.cloud import automl
+
+    # TODO(developer): Uncomment and saet the following variables
```
Comment (sirtorry) [inline automl/cloud-client/get_operation_status.py]: typo: saet

Comment (nnegrey) [reply automl/cloud-client/get_operation_status.py]: :( fixed

File: automl/cloud-client/language_entity_extraction_predict.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+    text_snippet = automl.types.TextSnippet(
+        content=content,
+        mime_type='text/plain')  # Types: 'text/plain', 'text/html'
+    payload = automl.types.ExamplePayload(text_snippet=text_snippet)
+
+    response = prediction_client.predict(model_full_id, payload)
+
+    for annotation_payload in response.payload:
+        print(u'Text Extract Entity Types: {}'.format(
```
Comment (sirtorry) [inline automl/cloud-client/language_entity_extraction_predict.py]: would it be possible for us to clarify what this is? display name of annotation payload  = text extract entity types could be confusing.

File: automl/cloud-client/vision_batch_predict.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    gcs_source = automl.types.GcsSource(
+        input_uris=[input_uri])
+
+    input_config = automl.types.BatchPredictInputConfig(gcs_source=gcs_source)
+    gcs_destination = automl.types.GcsDestination(
+        output_uri_prefix=output_uri)
+    output_config = automl.types.BatchPredictOutputConfig(
+        gcs_destination=gcs_destination)
+    # [0.0-1.0] Only produce results higher than this value
+    params = {'score_threshold': '0.8'}
```
Comment (sirtorry) [inline automl/cloud-client/vision_batch_predict.py]: vision and language batch predict have the same code. these optional params aren't unique to vision batch predict either.

Comment (nnegrey) [reply automl/cloud-client/vision_batch_predict.py]: Oh sweet!!! Didn't notice, thanks, I'll create a new just batch_predict.py file and then update docs, then delete the two different ones.

File: automl/cloud-client/language_entity_extraction_predict.py (hunk 5 above)
Comment (nnegrey) [reply automl/cloud-client/language_entity_extraction_predict.py]: Yea, I just copied the old samples on this one.
I'll try to put in something better.

File: automl/cloud-client/README.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+# AutoML Samples
+
+<a href="https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=automl/cloud-client/README.md">
+<img alt="Open in Cloud Shell" src ="http://gstatic.com/cloudssh/images/open-btn.png"></a>
+
+
+This directory contains samples for the [Google Cloud AutoML APIs](https://cloud.google.com/automl/) - [docs](https://cloud.google.com/automl/docs/)
+
+We highly reccommend that you refer to the official documentation pages:
```
Comment (texasmichelle) [inline automl/cloud-client/README.md]: s/reccommend/recommend/

File: automl/cloud-client/README.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  * [Entity Extraction](https://cloud.google.com/natural-language/automl/entity-analysis/docs)
+  * [Sentiment Analysis](https://cloud.google.com/natural-language/automl/sentiment/docs)
+* [AutoML Translation](https://cloud.google.com/translate/automl/docs)
+* AutoML Vision
+  * [Classification](https://cloud.google.com/vision/automl/docs)
+  * [Object Detection](https://cloud.google.com/vision/automl/object-detection/docs)
+
+This API is part of the larger collection of Cloud Machine Learning APIs.
+
+These Python samples demonstrates how to access the Cloud AutoML API
```
Comment (texasmichelle) [inline automl/cloud-client/README.md]: s/demonstrates/demonstrate/

File: automl/cloud-client/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+using the [Google Cloud Client Library for Python][google-cloud-python].
+
+[google-cloud-python]: https://github.com/GoogleCloudPlatform/google-cloud-python
+
+
+## Sample Types
+There are two types of samples: Base and API Specific
```
Comment (texasmichelle) [inline automl/cloud-client/README.md]: Add a period.

File: automl/cloud-client/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+### Model Management
+* [List Models](list_models.py)
+* [List Model Evaluation](list_model_evaluations.py) - For each AutoML Type the `metrics` field inside the model is unique, therefore each AutoML Type will have a
+small section of code to print out the `metrics` field. 
+* [Get Model](get_model.py)
+* [Get Model Evaluation](get_model_evaluation.py) - For each AutoML Type the `metrics` field inside the model is unique, therefore each AutoML Type will have a
+small section of code to print out the `metrics` field. 
+* [Delete Model](delete_model.py)
+* [Deploy Model](deploy_model.py) - Not supported by Translation
+* [Uneploy Model](undeploy_model.py) - Not supported by Translation
```
Comment (texasmichelle) [inline automl/cloud-client/README.md]: s/Uneploy/Unemploy/

File: automl/cloud-client/get_operation_status.py [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+    client = automl.AutoMlClient()
+    # Get the latest state of a long-running operation.
+    response = client.transport._operations_client.get_operation(
+        operation_full_id
+    )
+
+    print(u'Name: {}'.format(response.name))
+    print('Operation details:')
+    print(response)
```
Comment (sirtorry) [inline automl/cloud-client/get_operation_status.py]: there has been a request to print more useful details about the current status.

Comment (nnegrey) [reply automl/cloud-client/get_operation_status.py]: One problem is the `metadata` field inside an operation doesn't contain much useful information.
https://github.com/googleapis/googleapis/blob/master/google/cloud/automl/v1/operations.proto
Export and Batch have info about the request, but nothing about the progress. 
Potentially there might be a `done` field when set to `true`, but that can be checked.

File: automl/cloud-client/vision_classification_create_model.py [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    # project_id = 'YOUR_PROJECT_ID'
+    # dataset_id = 'YOUR_DATASET_ID'
+    # display_name = 'YOUR_MODEL_NAME'
+
+    client = automl.AutoMlClient()
+
+    # A resource that represents Google Cloud Platform location.
+    project_location = client.location_path(project_id, 'us-central1')
+    # Leave model unset to use the default base model provided by Google
+    metadata = automl.types.ImageClassificationModelMetadata()
```
Comment (sirtorry) [inline automl/cloud-client/vision_classification_create_model.py]: let's add train budget for all (possible) model creation samples

Comment (nnegrey) [reply automl/cloud-client/vision_classification_create_model.py]: it is an optional field, I'm not sure it is worth adding.

**Comment by texasmichelle** [comment]
Can you list the branches these came from?

**Comment by nnegrey** [comment]
None, they are written from scratch for the GAs.

**Comment by nnegrey** [comment]
Breaking up into smaller PRs, leave branch until docs are updated to master once smaller PRs are merged.

**Comment by nnegrey** [comment]
Please don't delete the branch, docs are using it.

**Pull request closed without merging**